### PR TITLE
Recsys overl

### PR DESCRIPTION
**Pull request by JKL98ISR**

### CODE REVIEW

File: deepchecks/recommender/suites.py [hunk 1]
```diff
@@ -14,27 +14,16 @@
...
-                                       DataDuplicates, DatasetsSizeComparison, DateTrainTestLeakageDuplicates,
-                                       DateTrainTestLeakageOverlap, FeatureDrift, FeatureFeatureCorrelation,
-                                       FeatureLabelCorrelation, FeatureLabelCorrelationChange,
-                                       IdentifierLabelCorrelation, IndexTrainTestLeakage, IsSingleValue, LabelDrift,
-                                       MixedDataTypes, MixedNulls, ModelInferenceTime, MultivariateDrift,
-                                       NewCategoryTrainTest, NewLabelTrainTest, OutlierSampleDetection, PercentOfNulls,
-                                       PredictionDrift, RegressionErrorDistribution, RocReport, SimpleModelComparison,
-                                       SingleDatasetPerformance, SpecialCharacters, StringLengthOutOfBounds,
-                                       StringMismatch, StringMismatchComparison, TrainTestPerformance,
-                                       TrainTestSamplesMix, UnusedFeatures, WeakSegmentsPerformance)
```
Comment (noamzbr) [inline deepchecks/recommender/suites.py:32]: Why not all tabular checks? Except the specific model evaluation ones such as CalibrationScore or BoostingOverfit

File: deepchecks/tabular/suite.py [hunk 2]
```diff
@@ -77,29 +82,105 @@ def run(
...
+            raise DeepchecksNotSupportedError('item_dataset is not supported for tabular datasets.')
+
+        if isinstance(train_dataset, RecDataset) and model_classes is not None:
+            raise DeepchecksNotSupportedError('model_classes is not supported for recommendation datasets.')
+    
         if self.context_type is None:
-            from deepchecks.recommender.context import Context as RecContext
-            from deepchecks.recommender.dataset import RecDataset
-            if train_dataset is None and isinstance(test_dataset, RecDataset) or isinstance(train_dataset, RecDataset):
+            if isinstance(train_dataset, RecDataset):
```
Comment (noamzbr) [inline deepchecks/tabular/suite.py:155]: Can we avoid this isinstance method?

File: deepchecks/recommender/suites.py (hunk 1 above)
Comment (JKL98ISR) [reply deepchecks/recommender/suites.py:32]: I copied from the prod suite

Comment (JKL98ISR) [reply deepchecks/recommender/suites.py:32]: if they want integrity let them run it alone

**Review by noamzbr — APPROVED** [review]

**Pull request merged**